packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -68,14 +69,40 @@ interface CommonProps
   placeholder?: string;
 
   /**
-   * An optional string prefix displayed before the input
+   * An optional prefix button displayed before the input
+   * Only one of PrefixButton, PrefixIcon, PrefixText can be set
    */
-  prefix?: string;
+  PrefixButton?: ReactNode;
 
   /**
-   * An optional string suffix displayed after the input
+   * An optional prefix icon displayed before the input
+   * Only one of PrefixButton, PrefixIcon, PrefixText can be set
    */
-  suffix?: string;
+  PrefixIcon?: ReactNode;
```

Comment:
@jeffbelser-okta - I couldn't figure out a typescript way to communicate that only one of the prefix/suffix could be set at one time. Do you have any suggestions?

Reply:
@sunnylittle-okta I think this would be done using a type union where member of the union would have a single type such as `prefixButton` or `prefixIcon`. Then in the component body we'll more than likely have to write some code to narrow types before we use any of the props which is OK too.

Reply:
would the `commonProps` include `PrefixButton` or `PrefixIcon` etc?
I tried different variations of union types but nothing would give me an error if I set both `PrefixButton` and PrefixIcon` on the props...which was the end result I preferred

Reply:
no I would think that common props could exclude those but we may need to tinker a bit with the exact type composition to get this working the way we want!